src/serac/physics/mesh.hpp

```diff
@@ -14,6 +14,7 @@
 
 #include "mfem.hpp"
 #include "serac/numerics/functional/tensor.hpp"
+#include "serac/numerics/functional/domain.hpp"
```

Comment:
The `struct Domain;` forward declaration wasn't enough, it seems, since when I included `mesh.hpp` in some of the tests, I got a bunch of errors complaining about `Domain`.